### PR TITLE
Address Open Issues on VAM Module

### LINKED ISSUE #116: Voters Account Management
In connection with the unaddressed issues from PR #107 

## Task List

- [x] #119
- [x] #118
- [x] #120
- [x] [While the email is sending, implement a button state for "Approve" and "Reject".](#issue4)

#### If you're unfamiliar on how to automatically mark or tick the checkbox, you can check this [video.](https://www.youtube.com/watch?v=d3N2yeAaLkc&t=391s)

### LINKED ISSUE #118: !IMPORTANT! Use prepared statements for every SQL Query.
Code Blocks:
`src/submission_handlers/manage-acc.php`
```php
$query = "SELECT * FROM voter WHERE account_status = 'for_verification' AND role = 'student_voter'";
``` 

`src/submission_handlers/move-to-trashbin.php`
```php
$inactive_query = "UPDATE voter SET account_status = 'invalid' WHERE voter_id = $voter_id"; 
``` 

`src/submission_handlers/update-status.php`
```php
    if ($status == 'Active') {
        $active_query = "UPDATE voter SET account_status = 'verified' WHERE voter_id = $voter_id";
        $voterManager->validateVoter($voter_id, $active_query);
    } elseif ($status == 'Disabled') {
        $inactive_query = "UPDATE voter SET account_status = 'Inactive' WHERE voter_id = $voter_id";
        $voterManager->validateVoter($voter_id, $inactive_query);
    } elseif ($status == 'Reject') {
        $for_verif_query = "UPDATE voter SET account_status = 'Rejected' WHERE voter_id = $voter_id";
        $voterManager->validateVoter($voter_id, $for_verif_query);
    }
```

`src/submission_handlers/validate-acc.php`
```php
    if ($action == 'approve') {
        // Updates status in database
        $approve_query = "UPDATE voter SET account_status = 'verified' WHERE voter_id = $voter_id";
        $voterManager->validateVoter($voter_id, $approve_query);

        // Sending of email
    } elseif ($action == 'reject') {

        // Updates status in database
        $reject_query = "UPDATE voter SET account_status = 'invalid' WHERE voter_id = $voter_id";
        $voterManager->validateVoter($voter_id, $reject_query);
```

### LINKED ISSUE #119: Fix deprecated datetime function in `manage-voters.php`
Code Block [line 373]:
```php
class=""><?php echo date("F j, Y", strtotime($row["status_updated"])); ?>
``` 
Error Logs:
```
2024-05-30 10:26 pm Error: [8192] strtotime(): Passing null to parameter #1 ($datetime) of type string is deprecated - C:\xampp\htdocs\PUPSRC-AutomatedElectionSystem\src\manage-voters.php: 373
```

### LINKED ISSUE #120: VAM: Redirection Defect
Fix defect of redirecting back to validate account page despite the account being validated already.

https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/162773681/e42cdeea-dc67-4090-8fec-980739edc8da

https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/162773681/ef9bc82d-59ea-4b8a-abd9-fa3a90a2aa42

### PR DESCRIPTION
**Pull request by biellamariscotes**
### Overview
This PR addresses the remaining issues opened at [PR #107](https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/pull/107).

<hr>

### Fixed Issues

5e463b06de1622f4d51f1db280f12d9e4f09b955 - Refactored the database query code to use prepared statements instead of direct queries.

878e25913f3ae7356e01bc397a40af0724633c68 
- Prevent access to voter details once the account has been deleted.
- Redirect to the `VAM table` if the back button is pressed after account deletion.
- Move invalid accounts to the `Recycle Bin` module for viewing.

fix #118
fix #119 
fix #120

### CODE REVIEW

**Comment by Carl-Tabuso** [comment]
To automatically close issue 118, 119, and 120, you need to use a closing keyword on your commit messages or in this pull request description.

image

Something like this:  
image

Upon closing of these issues, the task list/checkboxes tracking these three issues will be ticked/checked. Keep the above in mind. Thanks

File: src/validate-voter.php [hunk 1]
```diff
@@ -16,8 +16,12 @@
 	$voter_query = "SELECT * FROM voter WHERE voter_id = $voter_id";
 	$result = $conn->query($voter_query);
 	$row = $result->fetch_assoc();
```
Comment (Carl-Tabuso) [inline src/validate-voter.php]: Use prepared statement next time even though this is not a form submission — just for consistency and a bit of false sense of security. LOL

**Review by Carl-Tabuso — COMMENTED** [review]
Can you identify the root cause of these errors?
```
2024-05-31 7:13 pm Error: [8192] DateTime::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated - C:\xampp\htdocs\PUPSRC-AutomatedElectionSystem\src\voter-details.php: 175
2024-05-31 7:13 pm Error: [8192] strtotime(): Passing null to parameter #1 ($datetime) of type string is deprecated - C:\xampp\htdocs\PUPSRC-AutomatedElectionSystem\src\voter-details.php: 213
2024-05-31 7:17 pm Error: [8192] DateTime::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated - C:\xampp\htdocs\PUPSRC-AutomatedElectionSystem\src\manage-voters.php: 381
2024-05-31 7:17 pm Error: [8192] DateTime::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated - C:\xampp\htdocs\PUPSRC-AutomatedElectionSystem\src\manage-voters.php: 381
```

Check this
image

In voter table, the status of the user with that email remains NULL. We can proceed with the merging of these changes but we have to retain the issue #119

To give more context, I only simulated the approval of the account, nothing else.

**Comment by biellamariscotes** [comment]
fix #118 fix #119 fix #120

> Can you identify the root cause of these errors?
> 
> ```
> 2024-05-31 7:13 pm Error: [8192] DateTime::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated - C:\xampp\htdocs\PUPSRC-AutomatedElectionSystem\src\voter-details.php: 175
> 2024-05-31 7:13 pm Error: [8192] strtotime(): Passing null to parameter #1 ($datetime) of type string is deprecated - C:\xampp\htdocs\PUPSRC-AutomatedElectionSystem\src\voter-details.php: 213
> 2024-05-31 7:17 pm Error: [8192] DateTime::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated - C:\xampp\htdocs\PUPSRC-AutomatedElectionSystem\src\manage-voters.php: 381
> 2024-05-31 7:17 pm Error: [8192] DateTime::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated - C:\xampp\htdocs\PUPSRC-AutomatedElectionSystem\src\manage-voters.php: 381
> ```
> 
> Check this image
> 
> In voter table, the status of the user with that email remains NULL. We can proceed with the merging of these changes but we have to retain the issue #119
> 
> To give more context, I only simulated the approval of the account, nothing else.


Regarding this issue, I haven't encountered any errors on my end. It might be compatibility issues, perhaps? I am currently using Microsoft Edge.

Take a look:

https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/8dd29e4d-12a9-4bee-add7-1011017eafb7

File: src/validate-voter.php (hunk 1 above)
Comment (biellamariscotes) [reply src/validate-voter.php]: Revised in newly pushed commit: 467bda108d7580333f2f997a729cbe8b934459eb

**Review by Carl-Tabuso — APPROVED** [review]
Fucking triggers. Kill me

**Pull request merged**